src/main.ts

```diff
@@ -1,21 +1,44 @@
 import * as core from "@actions/core";
-const { GitHub, context } = require("@actions/github");
+import { getOctokit, context } from "@actions/github";
 import parse from "parse-diff";
 import { rexify } from "./utils";
 
+async function getDiff(octokit, context) {
+  const owner = context?.payload?.repository?.owner?.login;
+  const repo = context?.payload?.repository?.name;
+  const pull_number = context?.payload?.pull_request?.number;
+  core.info(`Getting diff for: ${owner}, ${repo}, ${pull_number}`);
+  const response = await octokit.rest.pulls.get({
+    owner,
+    repo,
+    pull_number,
+    headers: { accept: "application/vnd.github.v3.diff" },
+  });
+
+  const diff = response.data as unknown as string;
+  return parse(diff);
+}
+
 async function run() {
   try {
     // get information on everything
     const token = core.getInput("github-token", { required: true });
-    const github = new GitHub(token, {});
-    const user = context.payload.pull_request.user.login;
-    // First check for waived users
-    const waivedUsers = core.getInput("waivedUsers") || ["dependabot[bot]"];
+    const octokit = getOctokit(token);
 
-    if (waivedUsers.includes(user)) {
-      core.warning(`⚠️ Not running this workflow for waived user «${user}»`);
-      return;
-    }
+    const senderInfo = context?.payload?.sender;
+    const senderName = senderInfo?.login;
```

Comment:
If senderInfo is null, maybe this should bail out right here, it's going to be impossible to follow...